bundles/core/org.eclipse.smarthome.core.transform/src/main/java/org/eclipse/smarthome/core/transform/AbstractFileTransformationService.java

```diff
@@ -49,16 +51,20 @@
  * @author Kai Kreuzer - File caching mechanism
  * @author Markus Rathgeb - Add locale provider support
  */
+@NonNullByDefault
 public abstract class AbstractFileTransformationService<T> implements TransformationService {
 
+    @NonNullByDefault({})
```

Comment:
The watchService member could be null, so why disable the null checks instead of changing the code little bit and use the correct nullable annotation?

We could change the initializeWatchService method with something like this one that returns a non null watch service (just a non tested example):

```Java
private synchronized WatchService getWatchService() throws IOException {
    WatchService watchService = this.watchService;
    if (watchService == null) {
        try {
            watchService = this.watchService = FileSystems.getDefault().newWatchService();
            watchSubDirectory("");
        } catch (IOException e) {
            logger.error("Unable to start transformation directory monitoring");
        }
    }
    return watchService;
}
```

This also identifies an potential error in the current code:
If the watchService is null in the transform method, the initialize watch service method is called.
If `FileSystems.getDefault().newWatchService();` throws an IOException this is hidden from the caller and the watchService is still `null`.
After that `getLocalizedProposedFilename` is called, that itself could call `watchSubDirectory` that requires a non null watch service...

The annotation change does not make the runtime behavior different then it is now, but perhaps we should try to improve the implementation and error handling in another PR after that one.

WDYT?

Reply:
Sorry, to get offtopic, but is there some guarantee that the transform method is not called multiple times from different thread? If this is potential possible, the watchService could be initialized twice...

Reply:
@triller-telekom I tried to harden the watch service usage: #5367

Reply:
very good finding and thanks for changing the ode to be more robust with your PR.

Reply:
The PR #5367 has been merged, would you like to rebase your branch and mark this variable as `Nullable`

Reply:
Done